eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -739,29 +778,12 @@ instance
     pure $ translateToShelleyLedgerStateFromUtxo transContext startEpochNo Byron.empty
 
   impSatisfyNativeScript providedVKeyHashes _txBody script = do
-    keyPairs <- gets impKeyPairs
-    let
-      satisfyMOf m Empty
-        | m <= 0 = Just mempty
-        | otherwise = Nothing
-      satisfyMOf m (x :<| xs) =
-        case satisfyScript x of
-          Nothing -> satisfyMOf m xs
-          Just kps -> do
-            kps' <- satisfyMOf (m - 1) xs
-            Just $ kps <> kps'
-      satisfyScript = \case
-        RequireSignature keyHash
-          | keyHash `Set.member` providedVKeyHashes -> Just mempty
-          | otherwise -> do
-              keyPair <- Map.lookup keyHash keyPairs
-              Just $ Map.singleton keyHash keyPair
-        RequireAllOf ss -> satisfyMOf (length ss) ss
-        RequireAnyOf ss -> satisfyMOf 1 ss
-        RequireMOf m ss -> satisfyMOf m ss
-        _ -> error "Impossible: All NativeScripts should have been accounted for"
-
-    pure $ satisfyScript script
+    case script of
+      RequireSignature keyHash -> impSatisfySignature keyHash providedVKeyHashes
+      RequireAllOf ss -> impSatisfyMNativeScripts (length ss) ss providedVKeyHashes _txBody
+      RequireAnyOf ss -> impSatisfyMNativeScripts 1 ss providedVKeyHashes _txBody
```

Comment:
```suggestion
      RequireAnyOf ss -> do
        m <- choose (1, length ss)
        impSatisfyMNativeScripts m ss providedVKeyHashes _txBody
```

Reply:
Although I guess in the real world you're incentivised to satisfy only one script